tracing-subscriber/src/fmt/span.rs

```diff
@@ -62,8 +63,56 @@ enum State {
     Empty(usize),
 }
 
+struct SpanStackElement {
+    id: Id,
+    duplicate: bool,
+}
+
+struct SpanStack {
+    stack: Vec<SpanStackElement>,
+    ids: HashSet<Id>,
+}
+
+impl SpanStack {
+    fn new() -> Self {
+        SpanStack {
+            stack: vec![],
+            ids: HashSet::new(),
+        }
+    }
+
+    fn push(&mut self, id: Id) {
+        let duplicate = self.ids.contains(&id);
+        if !duplicate {
+            self.ids.insert(id.clone());
+        }
+        self.stack.push(SpanStackElement { id, duplicate })
+    }
+
+    fn pop(&mut self, expected_id: &Id) -> Option<Id> {
+        if self.stack.last().map(|elt| &elt.id) == Some(expected_id) {
```

Comment:
Nit, TIOLI: why not
```suggestion
        if self.stack.last()?.id == expected_id {
```

Reply:
nice :) I'm really not used to use the `?` operator on `Option` :)